src/main/java/baritone/command/defaults/ToMapCommand.java

```diff
@@ -0,0 +1,340 @@
+//            System.out.println("cr is null: " + (cr == null));
+//            if (cr == null)
+//                continue;
+                Map<String, int[]> cornerBlockCoordinatesForZeroZero = getBlockCoordsOfRegionCorners(rX, rZ);
+                int[] nw = cornerBlockCoordinatesForZeroZero.get("NW");
+                Map<Integer, Map<Integer, String>> topView = new LinkedHashMap<>(); // in block coordinates
+                Pattern p = Pattern.compile("Block\\{minecraft:(.*)\\}");
+                logDirect("Getting top blocks for region " + rX + " " + rZ);
+                for (int x = nw[0]; x < nw[0] + 512; x++) {
+                    if (!topView.containsKey(x)) {
+                        HashMap<Integer, String> zId = new LinkedHashMap<>();
+                        topView.put(x, zId);
+                    }
+                    for (int z = nw[1]; z < nw[1] + 512; z++) {
+                        for (int y = 256; y >= 0; y--) {
+                            try {
+                                BlockState bs = cr.getBlock(x, y, z);
+                                String id;
+                                if (bs == null) {
+                                    continue;
+//                                id = DEFAULT_BLOCK_ID; // does anyone have a better idea?
+                                } else {
+                                    try {
+                                        Matcher m = p.matcher(bs.toString());
+                                        if (m.find())
+                                            id = m.group(1);
+                                        else {
+                                            System.out.println("Failed to find block id for " + bs.toString());
+                                            return;
+                                        }
+                                    } catch (Exception e) {
+                                        System.out.println("No match found for " + bs.toString());
+                                        logDirect("No match found for " + bs.toString());
+                                        return;
+                                    }
+                                }
+                                if (bs == null || !(bs.getBlock() instanceof AirBlock)) { // we must show this block
+                                    topView.get(x).put(z, id);
+                                    break;
+                                }
+                            } catch (Exception e) {
+//                            System.out.println("Couldn't get BlockState for " + x + " " + y + " " + z);
+                            }
+                        }
+                        if (!topView.containsKey(x)) {
+                            if (!topView.get(x).containsKey(z)) {
+                                topView.get(x).put(z, DEFAULT_BLOCK_ID); // this is definitely not accurate
+                            }
+                        }
```

Comment:
Here's the culprit: You are only trying to insert missing z-entries when the whole x-row is missing, which is guaranteed to be false.
```suggestion
                        if (!topView.get(x).containsKey(z)) {
                            topView.get(x).put(z, DEFAULT_BLOCK_ID); // this is definitely not accurate
                        }
```

Reply:
Damn. Sorry, I'm not following but I have a feeling you found it problem

Reply:
This does not fix the missing block states but it does fix the loop range. Basically whenever we don't get a blockstate that entry is missing in topView.get(x) and the code you had there wasn't fixing this, so when we later use the entries to find the minimum z position there is no key lower that Interger.MAX_VALUE, because there is no key at all.